---
 setup.py |    4 ----
 1 file changed, 4 deletions(-)

--- wicd.orig/setup.py
+++ wicd/setup.py
@@ -467,15 +467,11 @@ try:
     (wpath.log, []), 
     (wpath.etc, []),
     (wpath.icons + 'scalable/apps/', ['icons/scalable/wicd-client.svg']),
-    (wpath.icons + '192x192/apps/', ['icons/192px/wicd-client.png']),
     (wpath.icons + '128x128/apps/', ['icons/128px/wicd-client.png']),
     (wpath.icons + '96x96/apps/', ['icons/96px/wicd-client.png']),
-    (wpath.icons + '72x72/apps/', ['icons/72px/wicd-client.png']),
     (wpath.icons + '64x64/apps/', ['icons/64px/wicd-client.png']),
     (wpath.icons + '48x48/apps/', ['icons/48px/wicd-client.png']),
-    (wpath.icons + '36x36/apps/', ['icons/36px/wicd-client.png']),
     (wpath.icons + '32x32/apps/', ['icons/32px/wicd-client.png']),
-    (wpath.icons + '24x24/apps/', ['icons/24px/wicd-client.png']),
     (wpath.icons + '22x22/apps/', ['icons/22px/wicd-client.png']),
     (wpath.icons + '16x16/apps/', ['icons/16px/wicd-client.png']),
     (wpath.images, [('images/' + b) for b in os.listdir('images') if not b.startswith('.')]),
