From: David Paleino <dapal@debian.org>
Subject: Handle API change between urwid 0.9.8.4 and urwid 0.9.9
Bug-Debian: http://bugs.debian.org/557462
Bug: https://bugs.launchpad.net/wicd/+bug/490416
Forwarded: yes,  https://bugs.launchpad.net/wicd/+bug/490416/comments/2

---
 curses/curses_misc.py |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

--- wicd.orig/curses/curses_misc.py
+++ wicd/curses/curses_misc.py
@@ -361,7 +361,11 @@ class ComboBox(urwid.WidgetWrap):
 
     def set_focus(self,index):
         self.focus = index
-        self.cbox.set_w(SelText(self.list[index]+self.DOWN_ARROW))
+        # API changed between urwid 0.9.8.4 and 0.9.9
+        try:
+            self.cbox.set_w(SelText(self.list[index]+self.DOWN_ARROW))
+        except AttributeError:
+            self.cbox._w = SelText(self.list[index]+self.DOWN_ARROW)
         if self.overlay:
             self.overlay._listbox.set_focus(index)
 
@@ -400,7 +404,11 @@ class ComboBox(urwid.WidgetWrap):
             retval = self.overlay.show(self.ui,self.parent)
             if retval != None:
                 self.set_focus(self.list.index(retval))
-                #self.cbox.set_w(SelText(retval+'    vvv'))
+                ## API changed between urwid 0.9.8.4 and 0.9.9
+                #try:
+                #    self.cbox.set_w(SelText(retval+'    vvv'))
+                #except AttributeError:
+                #    self.cbox._w = SelText(retval+'    vvv')
                 if self.callback != None:
                     self.callback(self,self.overlay._listbox.get_focus()[1],
                             self.user_args)
