From: David Paleino <dapal@debian.org>
Subject: fix minor issue with help message -- remove .py
Forwarded: no

---
 curses/wicd-curses.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- wicd.orig/curses/wicd-curses.py
+++ wicd/curses/wicd-curses.py
@@ -1026,7 +1026,7 @@ setup_dbus()
 ########################################
 if __name__ == '__main__':
     try:
-        parser = OptionParser(version="wicd-curses-%s (using wicd %s)" % (CURSES_REV,daemon.Hello()))
+        parser = OptionParser(version="wicd-curses-%s (using wicd %s)" % (CURSES_REV,daemon.Hello()), prog="wicd-curses")
     except Exception, e:
         if "DBus.Error.AccessDenied" in e.get_dbus_name():
             print language['access_denied_wc'].replace('$A','\033[1;34m'+wpath.wicd_group+'\033[0m')
