From: David Paleino <dapal@debian.org>
Subject: hopefully the last bits remaining from the codesplit

---
 gtk/configscript.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- wicd.orig/gtk/configscript.py
+++ wicd/gtk/configscript.py
@@ -155,7 +155,7 @@ def main (argv):
     
     script_info = get_script_info(network, network_type)
     
-    gladefile = wpath.share + "wicd.glade"
+    gladefile = wpath.gtk + "wicd.glade"
     wTree = gtk.glade.XML(gladefile)
     dialog = wTree.get_widget("configure_script_dialog")
     wTree.get_widget("pre_label").set_label(language['before_script'] + ":")
