From: David Paleino <dapal@debian.org>
Subject: fix wrong detection of udhcpc failing
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=563619
Forwarded: no

---
 wicd/wnettools.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- wicd.orig/wicd/wnettools.py
+++ wicd/wicd/wnettools.py
@@ -561,7 +561,7 @@ class BaseInterface(object):
 
         while not udhcpc_complete:
             line = pipe.readline()
-            if line.endswith("failing"):
+            if line.endswith("failing."):
                 udhcpc_success = False
                 udhcpc_complete = True
             elif line == '':
