From: David Paleino <dapal@debian.org>
Subject: make network properties passed on command-line case-insensitive
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=563620
Forwarded: no

---
 cli/wicd-cli.py |    1 +
 1 file changed, 1 insertion(+)

--- wicd.orig/cli/wicd-cli.py
+++ wicd/cli/wicd-cli.py
@@ -143,6 +143,7 @@ if options.network_details:
 # network properties
 
 if options.network_property:
+	options.network_property = option.network_property.lower()
 	if options.wireless:
 		is_valid_wireless_network_id(options.network)
 		if not options.set_to:
