From: David Paleino <dapal@debian.org>
Subject: default to current active wireless network when querying for a
 property
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=563622
Forwarded: no

---
 cli/wicd-cli.py |   11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

--- wicd.orig/cli/wicd-cli.py
+++ wicd/cli/wicd-cli.py
@@ -145,11 +145,16 @@ if options.network_details:
 if options.network_property:
 	options.network_property = option.network_property.lower()
 	if options.wireless:
-		is_valid_wireless_network_id(options.network)
+		if options.network >= 0:
+			is_valid_wireless_network_id(options.network)
+			network_id = options.network
+		else:
+			network_id = wireless.GetCurrentNetworkID(0)
+			is_valid_wireless_network_id(network_id)
 		if not options.set_to:
-			print wireless.GetWirelessProperty(options.network, options.network_property)
+			print wireless.GetWirelessProperty(network_id, options.network_property)
 		else:
-			wireless.SetWirelessProperty(options.network, \
+			wireless.SetWirelessProperty(network_id, \
 					options.network_property, options.set_to)
 	elif options.wired:
 		if not options.set_to:
