From: David Paleino <dapal@debian.org>
Subject: don't execute hidden or backup files in scripts directories
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=563553
Forwarded: no

---
 wicd/misc.py |    2 ++
 1 file changed, 2 insertions(+)

--- wicd.orig/wicd/misc.py
+++ wicd/wicd/misc.py
@@ -186,6 +186,8 @@ def ExecuteScripts(scripts_dir, verbose=
     if not os.path.exists(scripts_dir):
         return
     for obj in sorted(os.listdir(scripts_dir)):
+        if re.match("^\..*", obj) or re.match(".*~$", obj):
+            continue
         obj = os.path.abspath(os.path.join(scripts_dir, obj))
         if os.path.isfile(obj) and os.access(obj, os.X_OK):
             ExecuteScript(os.path.abspath(obj), verbose=verbose,
