From: David Paleino <dapal@debian.org>
Subject: don't make dhcpcd handle link-local
Forwarded: no

---
 wicd/wnettools.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- wicd.orig/wicd/wnettools.py
+++ wicd/wicd/wnettools.py
@@ -292,7 +292,7 @@ class BaseInterface(object):
                   'id' : misc.PUMP,
                 },
             "dhcpcd" : 
-                {'connect' : r"%(cmd)s %(iface)s -h %(hostname)s ",
+                {'connect' : r"%(cmd)s %(iface)s -h %(hostname)s --noipv4ll ",
                  'release' : r"%(cmd)s -k %(iface)s",
                  'id' : misc.DHCPCD,
                 },
