From: David Paleino <dapal@debian.org>
Subject: fix deprecation warnings
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=571314
Forwarded: no

---
 gtk/gui.py |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- wicd.orig/gtk/gui.py
+++ wicd/gtk/gui.py
@@ -430,7 +430,7 @@ class appGui(object):
             gobject.idle_add(self.all_network_list.set_sensitive, True)
             gobject.idle_add(self.status_area.hide_all)
         if self.statusID:
-            gobject.idle_add(self.status_bar.remove, 1, self.statusID)
+            gobject.idle_add(self.status_bar.remove_message, 1, self.statusID)
     
     def set_connecting_state(self, info):
         if not self.connecting:
@@ -445,7 +445,7 @@ class appGui(object):
             gobject.idle_add(self.all_network_list.set_sensitive, False)
             gobject.idle_add(self.status_area.show_all)
         if self.statusID:
-            gobject.idle_add(self.status_bar.remove, 1, self.statusID)
+            gobject.idle_add(self.status_bar.remove_message, 1, self.statusID)
         if info[0] == "wireless":
             gobject.idle_add(self.set_status, str(info[1]) + ': ' +
                    language[str(wireless.CheckWirelessConnectingMessage())])
@@ -698,7 +698,7 @@ class appGui(object):
             cancel_button.set_sensitive(True)
             self.all_network_list.set_sensitive(False)
             if self.statusID:
-                gobject.idle_add(self.status_bar.remove, 1, self.statusID)
+                gobject.idle_add(self.status_bar.remove_message, 1, self.statusID)
             gobject.idle_add(self.set_status, language["disconnecting_active"])
             gobject.idle_add(self.status_area.show_all)
             self.wait_for_events()
