From: Marin Ivanov <metala@metala.org>
Subject: wait for the child process to end
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=537195
Forwarded: no

---
 wicd/wnettools.py |    1 +
 1 file changed, 1 insertion(+)

--- wicd.orig/wicd/wnettools.py
+++ wicd/wicd/wnettools.py
@@ -619,6 +619,7 @@ class BaseInterface(object):
         else:
             print "ERROR: no dhcp client found"
             ret = None
+        self.dhcp_object.wait()
         return ret
         
     @neediface(False)
