From: David Paleino <dapal@debian.org>
Subject: support /etc/network/ hierarchy for connection scripts
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=579497
Forwarded: no

---
 wicd/networking.py |   24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

--- wicd.orig/wicd/networking.py
+++ wicd/wicd/networking.py
@@ -210,6 +210,12 @@ class Controller(object):
             mac = 'X'
         if name in (None, ''):
             name = 'X'
+        os.putenv("MODE", "stop")
+        os.putenv("VERBOSITY", int(self.debug))
+        os.putenv("IFACE", iface)
+        os.putenv("LOGICAL", iface)
+        os.putenv("PHASE", "pre-down")
+        misc.ExecuteScripts('/etc/network/if-down.d/', self.debug)
         misc.ExecuteScripts(wpath.predisconnectscripts, self.debug,
                            extra_parameters=(nettype, name, mac))
         if self.pre_disconnect_script:
@@ -223,6 +229,8 @@ class Controller(object):
         iface.FlushRoutes()
         iface.Down()
         iface.Up()
+        os.putenv("PHASE", "post-down")
+        misc.ExecuteScripts('/etc/network/if-post-down.d/', self.debug)
         misc.ExecuteScripts(wpath.postdisconnectscripts, self.debug,
                             extra_parameters=(nettype, name, mac))
         if self.post_disconnect_script:
@@ -840,6 +848,12 @@ class WirelessConnectThread(ConnectThrea
         self.is_connecting = True
         
         # Run pre-connection script.
+        os.putenv("MODE", "start")
+        os.putenv("VERBOSITY", int(self.debug))
+        os.putenv("IFACE", wiface)
+        os.putenv("LOGICAL", wiface)
+        os.putenv("PHASE", "pre-up")
+        self.run_global_scripts_if_needed('/etc/network/if-pre-up.d/')
         self.run_global_scripts_if_needed(wpath.preconnectscripts,
                                           extra_parameters=('wireless',
                                                     self.network['essid'],
@@ -889,6 +903,8 @@ class WirelessConnectThread(ConnectThrea
         self.verify_association(wiface)
         
         # Run post-connection script.
+        os.putenv("PHASE", "post-up")
+        self.run_global_scripts_if_needed('/etc/network/if-up.d/')
         self.run_global_scripts_if_needed(wpath.postconnectscripts,
                                           extra_parameters=('wireless',
                                                     self.network['essid'],
@@ -1098,6 +1114,12 @@ class WiredConnectThread(ConnectThread):
         self.is_connecting = True
 
         # Run pre-connection script.
+        os.putenv("MODE", "start")
+        os.putenv("VERBOSITY", int(self.debug))
+        os.putenv("IFACE", liface)
+        os.putenv("LOGICAL", liface)
+        os.putenv("PHASE", "pre-up")
+        self.run_global_scripts_if_needed('/etc/network/if-pre-up.d/')
         self.run_global_scripts_if_needed(wpath.preconnectscripts,
                                           extra_parameters=('wired', 'wired',
                                                             self.network['profilename'])
@@ -1120,6 +1142,8 @@ class WiredConnectThread(ConnectThread):
         self.set_dns_addresses(liface)
         
         # Run post-connection script.
+        os.putenv("PHASE", "post-up")
+        self.run_global_scripts_if_needed('/etc/network/if-up.d/')
         self.run_global_scripts_if_needed(wpath.postconnectscripts,
                                           extra_parameters=('wired', 'wired',
                                                 self.network['profilename'])
