From: David Paleino <dapal@debian.org>
Subject: make info in "Connection Info" dialog selectable (i.e. for copy&paste)
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=571579
Forwarded: no

---
 gtk/wicd-client.py |    1 +
 1 file changed, 1 insertion(+)

--- wicd.orig/gtk/wicd-client.py
+++ wicd/gtk/wicd-client.py
@@ -550,6 +550,7 @@ class TrayIcon(object):
             # Create labels
             self.label = gtk.Label()
             self.data = gtk.Label()
+            self.data.set_selectable(True)
             self.label.show()
             self.data.show()
             self.list = []
