From: David Paleino <dapal@debian.org>
Subject: generalize libc linking using ctypes
Bug-Debian: http://bugs.debian.org/636941
Forwarded: no

---
 wicd/misc.py |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

--- wicd.orig/wicd/misc.py
+++ wicd/wicd/misc.py
@@ -406,11 +406,8 @@ def RenameProcess(new_name):
         return False
     try:
         import ctypes
-        is_64 = os.path.exists('/lib64/libc.so.6')
-        if is_64:
-            libc = ctypes.CDLL('/lib64/libc.so.6')
-        else:
-            libc = ctypes.CDLL('/lib/libc.so.6')
+        from ctypes.util import find_library
+        libc = ctypes.CDLL(find_library('c'))
         libc.prctl(15, new_name, 0, 0, 0)
         return True
     except:
