From: David Paleino <dapal@debian.org>
Subject: fix channel parsing
Bug-Debian: http://bugs.debian.org/640075
Bug-Ubuntu: https://bugs.launchpad.net/wicd/+bug/545248
Origin: upstream, http://bazaar.launchpad.net/~wicd-devel/wicd/experimental/revision/581

---
 wicd/wnettools.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- wicd.orig/wicd/wnettools.py
+++ wicd/wicd/wnettools.py
@@ -45,7 +45,7 @@ from misc import find_path
 _re_mode = (re.I | re.M | re.S)
 essid_pattern = re.compile('.*ESSID:"?(.*?)"?\s*\n', _re_mode)
 ap_mac_pattern = re.compile('.*Address: (.*?)\n', _re_mode)
-channel_pattern = re.compile('.*Channel:?=? ?(\d\d?)', _re_mode)
+channel_pattern = re.compile('.*Channel:?=? ?(\d+)', _re_mode)
 strength_pattern = re.compile('.*Quality:?=? ?(\d+)\s*/?\s*(\d*)', _re_mode)
 altstrength_pattern = re.compile('.*Signal level:?=? ?(\d+)\s*/?\s*(\d*)', _re_mode)
 signaldbm_pattern = re.compile('.*Signal level:?=? ?(-\d\d*)', _re_mode)
